src/swoop/api/config.py

```diff
@@ -10,7 +12,43 @@ class Settings(BaseSettings):
     database_user: str
     database_pass: str
     database_name: str
-    database_url: str
+    database_url_extra: str = ""
+
+    db_min_conn_size: int = 2
+    db_max_conn_size: int = 2
+    db_max_queries: int = 50000
+    db_max_inactive_conn_lifetime: float = 300
+
+    def build_db_connection_string(self, **kwargs):
+        """Build a DB connection string from setttings, with optional overrides"""
+        ctx = {
+            "host": self.database_host,
+            "port": self.database_port,
+            "user": self.database_user,
+            "password": self.database_pass,
+            "name": self.database_name,
+            "url_extra": self.database_url_extra,
+        }
+        ctx.update(kwargs)
+        return (
+            f"postgresql://{ctx['user']}:{quote(ctx['password'])}"
+            f"@{ctx['host']}:{ctx['port']}/{ctx['name']}{ctx['url_extra']}"
+        )
+
+    @property
+    def reader_connection_string(self):
+        """Create reader psql connection string."""
+        return self.build_db_connection_string()
```

Comment:
Going with [this comment](https://github.com/Element84/swoop/pull/7/files#r1176766708), I think this and the `writer_connection_string` might allow a user to specify a `database_url` to override the builder method, perhaps like this:
```suggestion
        return self.database_url if hasattr(self, 'database_url') else self.build_db_connection_string()
```

I would not want to add `database_url` back in to the `Settings` model, as I think this would simply be for testing and we wouldn't want to expose such a setting to users.